test/sync/src/androidTest/kotlin/io/realm/test/mongodb/shared/internal/KtorNetworkTransportTest.kt

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2021 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.test.mongodb.shared.internal
+
+import io.realm.interop.Response
+import io.realm.mongodb.internal.KtorNetworkTransport
+import io.realm.internal.platform.singleThreadDispatcher
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+import kotlin.test.assertEquals
+
+private const val BASE_URL = "http://127.0.0.1:8888" // URL to command server
+
+internal class KtorNetworkTransportTest {
+
+    private lateinit var transport: KtorNetworkTransport
+
+    enum class HTTPMethod(val nativeKey: String) {
+        GET("get"),
+        POST("post"),
+        PATCH("patch"),
+        PUT("put"),
+        DELETE("delete")
+    }
+
+    @BeforeTest
+    fun setUp() {
+        transport = KtorNetworkTransport(
+            timeoutMs = 5000,
+            dispatcher = singleThreadDispatcher("ktor-test")
+        )
+    }
+
+    @Test
+    fun requestSuccessful() {
+        val url = "$BASE_URL/okhttp?success=true"
+
+        // FIXME test fails on native due to method ordering, e.g.:
```

Comment:
I believe this was fixed and was a problem with Ktor setup?

Reply:
Works with the `ktor-client-cio` for `macos`. Don't really know why, as the client isn't listed for native desktop.